mne/io/eeglab/eeglab.py

```diff
@@ -369,10 +689,18 @@ def __init__(self, input_fname, montage, eog=(), event_id=None,
                      'the .set file')
             # can't be done in standard way with preload=True because of
             # different reading path (.set file)
+
             if eeg.nbchan == 1 and len(eeg.data.shape) == 1:
                 n_chan, n_times = [1, eeg.data.shape[0]]
             else:
                 n_chan, n_times = eeg.data.shape
+
+            # Seem to have transpose with matlab hdf storage
+            if hdf5_transpose:
+                temp = eeg.data.transpose()
+                eeg.data = temp
```

Comment:
same thing -- if data needs to be transposed, can you do it in the above function so that this one can stay "clean"?